src/esp/gfx/TextureVisualizerShader.h

```diff
@@ -46,12 +46,28 @@ class TextureVisualizerShader : public Magnum::GL::AbstractShaderProgram {
   /** @brief Constructor */
   explicit TextureVisualizerShader(Flags flags);
 
+  /**
+   * @brief Build the colorMapTexture_ based on the passed @p colorMap of
+   * colors.
+   *
+   * @param colorMap The map of colors to use
+   * @param wrapEdges Whether queries exceeding the length of @p colorMap will
+   * use the final color, or will wrap around on the texture.
+   * @return Reference to self (for method chaining)
+   */
+  TextureVisualizerShader& setColorMapTexture(
+      Corrade::Containers::ArrayView<const Magnum::Vector3ub> colorMap,
+      bool wrapEdges,
```

Comment:
[Boolean function parameters are an API smell](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap) in languages that don't have the concept of named parameters. I'd expose the `Mn::SamplerWrapping` type directly.

I'm also not sure about the offset/scale parameters. Why not just let the caller call also `setColorMapTransformation()`? There the parameters are documented and the responsibility/interaction is clear.

Reply:
> I'm also not sure about the offset/scale parameters. Why not just let the caller call also `setColorMapTransformation()`? There the parameters are documented and the responsibility/interaction is clear, now it's a bit unclear how the two functions affect each other.

Originally the code was set up to fully instantiate the ShaderProgram through the constructor (including setting these values) instead of having a secondary call to setColorMapTransformation for these individual values. I tried to keep my refactoring to the existing code as minimal as possible, but if you think this should be changed, i'm all for it.